crypto/merkle/proof.go

```diff
@@ -43,6 +43,9 @@ func (poz ProofOperators) Verify(root []byte, keypath string, args [][]byte) (er
 	for i, op := range poz {
 		key := op.GetKey()
 		if len(key) != 0 {
+			if len(keys) == 0 {
```

Comment:
😼 https://github.com/tendermint/tendermint/pull/2733/files#r229621169

Reply:
Oh damn, I was obviously wrong, since the keys were getting re-sliced and could run out before we finish looping over the poz. Very nice catch!